pkg/resource/deploy/step_generator.go

```diff
@@ -853,6 +853,103 @@ func processIgnoreChanges(inputs, oldInputs resource.PropertyMap,
 	return ignoredInputs.ObjectValue(), nil
 }
 
+func copyWithUnknownElements(oldValue resource.PropertyValue) resource.PropertyValue {
+	unknown := resource.MakeComputed(resource.NewStringProperty(""))
+	switch {
+	case oldValue.IsObject():
+		newValue := resource.PropertyMap{}
+		for k := range oldValue.ObjectValue() {
+			newValue[k] = unknown
+		}
+		return resource.NewObjectProperty(newValue)
+	case oldValue.IsArray():
+		newValue := make([]resource.PropertyValue, len(oldValue.ArrayValue()))
+		for i := range newValue {
+			newValue[i] = unknown
+		}
+		return resource.NewArrayProperty(newValue)
+	default:
+		return unknown
+	}
+}
+
+func copyStableValue(oldState, newState resource.PropertyMap, path resource.PropertyPath, allowPartialValues bool) {
+	if len(path) == 0 || len(path) > 1 && !allowPartialValues {
+		return
+	}
+
+	oldParent, newParent := resource.NewObjectProperty(oldState), resource.NewObjectProperty(newState)
+	for _, element := range path[:len(path)-1] {
+		switch element := element.(type) {
+		case string:
+			if !oldParent.IsObject() || !newParent.IsObject() {
+				return
+			}
+			key := resource.PropertyKey(element)
+			old, ok := oldParent.ObjectValue()[key]
+			if !ok {
+				return
+			}
+			new, ok := newParent.ObjectValue()[key]
+			if !ok || new.IsComputed() {
+				new = copyWithUnknownElements(old)
```

Comment:
so i somewhat understand what's going on here.  either we didnt' have the value, or it was actually computed...  the part i'm not getting is why we do `copyWithUnknownElements` causing a map to go to map and an array to an array (just with unknown elements).  Can you document here (or on copyWithUnknownElements) what the thought is here?